src/background/handler.js

```
@@ -0,0 +1,74 @@
+ * @param {chrome.runtime.MessageSender} sender - Metadata about the script that sent the message.
+ * @param {function} sendResponse - Function to call to send a response.
+ */
+function handleReceiveMessage(message, sender, sendResponse) {
+    let response = {error: '', result: {}} // Initialize response object.
+
+    try {
+        // Switch based on the method specified in the message.
+        switch (message.method) {
+            case 'GetEggByDescription':
+                // Parses the sender's URL to extract a pathname component.
+                let url = new URL(sender.url);
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)